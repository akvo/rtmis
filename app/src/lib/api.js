import axios from 'axios';

export const config = {
  baseURL: null,
  headers: {
    'Content-Type': 'application/json',
  },
};

const API = () => {
  const getConfig = () => {
    let current = config;
    if (api.baseURL) {
      current = {
        ...config,
        baseURL: api.baseURL,
      };
    }
    return api?.token
      ? {
          ...current,
          headers: { ...config.headers, Authorization: `Bearer ${api.token}` },
        }
      : current;
  };
  return {
    get: (url, config = {}) => axios({ url, ...getConfig(), ...config }),
    post: (url, data, config = {}) =>
      axios({
        url,
        method: 'POST',
        data,
        ...{
          ...getConfig(),
          headers: {
            ...getConfig().headers,
            ...config?.headers,
          },
        },
      }),
    put: (url, data, config) => axios({ url, method: 'PUT', data, ...getConfig(), ...config }),
    patch: (url, data, config) => axios({ url, method: 'PATCH', data, ...getConfig(), ...config }),
    delete: (url) => axios({ url, method: 'DELETE', ...getConfig() }),
    setToken: (token) => {
      api.token = token;
    },
    setServerURL: (serverURL) => {
      api.baseURL = serverURL;
    },
    getConfig,
  };
};

const api = API();

export default api;
