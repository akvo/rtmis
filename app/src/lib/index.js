export { default as api } from './api';
export { default as loc } from './loc';
export { default as notification } from './notification';
export { default as backgroundTask } from './background-task';
export { default as i18n } from './i18n';
export { default as cascades } from './cascades';
