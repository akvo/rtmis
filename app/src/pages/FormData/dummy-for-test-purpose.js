export const washInSchool = {
  createdAt: '2023-08-03T10:48:49.470Z',
  duration: 3,
  form: 3,
  geo: '-7.391187|109.4651083',
  id: 5,
  json: '"{\\"1681103820975\\":65,\\"1681107307245\\":\\"Test Village\\",\\"1681107334328\\":[-7.391187,109.4651083],\\"1681178759979\\":[\\"Yes\\"],\\"1681108078143\\":\\"Test scool\\",\\"1681108101856\\":[\\"Rural\\"],\\"1681108181965\\":[\\"Classic\\"],\\"1681107224764\\":[\\"Yes\\"],\\"1681108436898\\":[\\"Pipeline connection / Piped water to yard/plot\\"],\\"1681108776850\\":[\\"Totally Functional\\"],\\"1681108950633\\":[\\"Yes\\"],\\"1681109061118\\":[\\"Yes\\"],\\"1681109157654\\":[\\"No\\"],\\"1681110132819\\":[\\"Clean\\"],\\"1681107239045\\":[\\"No\\"],\\"1681110373320\\":[\\"In the free area\\"],\\"1681107334444\\":\\"2021-12-22\\",\\"1681107335555\\":\\"data:image/jpeg;base64,dummyR0ndomb5e64\\",\\"1681107261379\\":[\\"No\\"]}"',
  name: 'Bagassi - Test Village',
  submitted: 1,
  submittedAt: '2023-08-03T10:48:49.470Z',
  syncedAt: '2023-08-03T10:48:56.159Z',
  user: 1,
};

export const washInSchoolForm = {
  id: 1681103820972,
  form: 'WASH in Schools',
  type: 1,
  languages: ['en', 'fr'],
  defaultLanguage: 'en',
  translations: [
    {
      language: 'fr',
      name: 'Questionnaire Ecoles',
    },
  ],
  question_group: [
    {
      id: 1681103820974,
      question_group: 'Localisation',
      order: 1,
      repeatable: false,
      question: [
        {
          id: 1681103820975,
          order: 1,
          questionGroupId: 1681103820974,
          question: 'Locality',
          type: 'cascade',
          source: {
            file: 'administration.sqlite',
            parent_id: 0,
          },
          required: true,
          meta: true,
          translations: [
            {
              language: 'fr',
              name: 'Localite',
            },
          ],
        },
        {
          id: 1681107307245,
          order: 2,
          questionGroupId: 1681103820974,
          question: 'Village',
          type: 'text',
          required: false,
          meta: true,
          translations: [
            {
              language: 'fr',
              name: 'Village',
            },
          ],
        },
        {
          id: 1681107334328,
          order: 3,
          questionGroupId: 1681103820974,
          question: 'School GPS coordinates',
          type: 'geo',
          required: true,
          meta: true,
          translations: [
            {
              language: 'fr',
              name: "Coordonnes GPS de l'école",
            },
          ],
        },
        {
          id: 1681107334444,
          order: 4,
          questionGroupId: 1681103820974,
          question: 'School registered date',
          type: 'date',
          required: true,
          meta: true,
          translations: [
            {
              language: 'fr',
              name: "Date d'inscription à l'école",
            },
          ],
        },
        {
          id: 1681107335555,
          order: 4,
          questionGroupId: 1681103820974,
          question: 'School photo',
          type: 'photo',
          required: true,
          meta: true,
          translations: [
            {
              language: 'fr',
              name: "Photo d'école",
            },
          ],
        },
      ],
      translations: [
        {
          language: 'fr',
          name: 'Localisation',
        },
      ],
    },
    {
      id: 1681178759978,
      question_group: 'Consent',
      order: 2,
      repeatable: false,
      question: [
        {
          id: 1681178759979,
          order: 1,
          questionGroupId: 1681078759978,
          question:
            'We carry out this interview to better understand the situation in terms of information and prevention on the COVVI-19 in your area. Do you willing to participate to participate?',
          type: 'option',
          required: true,
          meta: false,
          translations: [
            {
              language: 'fr',
              name: 'Nous réalisons cet entretien pour mieux comprendre la situation en matière d’information et de prévention sur la COVID-19 dans votre zone.Acceptez vous volontier de participer?',
            },
          ],
          options: [
            {
              id: 1681078772186,
              code: null,
              name: 'Yes',
              order: 1,
              translations: [
                {
                  language: 'fr',
                  name: 'Oui',
                },
              ],
            },
            {
              id: 1681078772187,
              code: null,
              name: 'No',
              order: 2,
              translations: [
                {
                  language: 'fr',
                  name: 'Non',
                },
              ],
            },
          ],
        },
      ],
      translations: [
        {
          language: 'fr',
          name: 'Consentement',
        },
      ],
    },
  ],
};
