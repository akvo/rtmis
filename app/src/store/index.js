export { default as AuthState } from './auth';
export { default as BuildParamsState } from './buildParams';
export { default as FormState } from './forms';
export { default as UIState } from './ui';
export { default as UserState } from './users';
