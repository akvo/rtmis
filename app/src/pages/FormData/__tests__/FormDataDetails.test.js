import React, { useState } from 'react';
import { render, renderHook, fireEvent, act, waitFor } from '@testing-library/react-native';
import { useNavigation, triggerBeforeRemoveEvent } from '@react-navigation/native';

jest.mock('@react-navigation/native');

import { FormState, UIState } from '../../../store';
import FormDataDetails from '../FormDataDetails';
import { washInSchool, washInSchoolForm } from '../dummy-for-test-purpose';
import { cascades } from '../../../lib';

jest.mock('expo-sqlite');
jest.mock('../../../lib', () => ({
  cascades: {
    loadDataSource: jest.fn(async (source, id) => {
      return id
        ? { rows: { length: 1, _array: [{ id: 65, name: 'Administration 65', parent: 0 }] } }
        : {
            rows: {
              length: 2,
              _array: [
                { id: 65, name: 'Administration 65', parent: 0 },
                { id: 66, name: 'Administration 66', parent: 0 },
              ],
            },
          };
    }),
  },
  i18n: {
    text: jest.fn(() => ({
      latitude: 'Latitude',
      longitude: 'Longitude',
    })),
  },
}));

describe('FormDataDetails', () => {
  beforeAll(() => {
    const { json: valuesJSON } = washInSchool;
    act(() => {
      UIState.update((s) => {
        s.lang = 'en';
      });
      FormState.update((s) => {
        s.form = {
          json: JSON.stringify(washInSchoolForm).replace(/'/g, "''"),
        };
        s.currentValues = JSON.parse(JSON.parse(valuesJSON));
      });
    });
  });

  it('should render correctly', async () => {
    const mockNavigation = useNavigation();
    const { result: resultState } = renderHook(() => useState(0));
    const { result: resultForm } = renderHook(() => FormState.useState((s) => s.form));
    const { result: resultValues } = renderHook(() => FormState.useState((s) => s.currentValues));
    const [currentPage, setCurrentPage] = resultState.current;
    const { json: formJSON } = resultForm.current;
    const values = resultValues.current;

    const { getByTestId, debug } = render(
      <FormDataDetails
        navigation={mockNavigation}
        formJSON={formJSON}
        values={values}
        currentPage={currentPage}
        setCurrentPage={setCurrentPage}
      />,
    );

    const { result: resultCascade } = renderHook(() => useState(null));
    const [cascadeValue, setCascadeValue] = resultCascade.current;

    act(() => {
      setCascadeValue({ id: 65, name: 'Administration 65', parent: 0 });
    });

    // Wait for the fetchCascade to complete its asynchronous behavior
    await waitFor(() => expect(cascades.loadDataSource).toHaveBeenCalledTimes(1));

    await waitFor(() => {
      expect(resultCascade.current[0]).toEqual({ id: 65, name: 'Administration 65', parent: 0 });
      const questionText = getByTestId('text-question-0');
      expect(questionText).toBeDefined();
      const answerText = getByTestId('text-answer-0');
      expect(answerText).toBeDefined();
    });
  });

  it('should match with snapshot', async () => {
    const mockNavigation = useNavigation();
    const mockRoute = {
      params: {
        name: 'Datapoint name',
      },
    };

    const tree = render(<FormDataDetails navigation={mockNavigation} route={mockRoute} />);
    await waitFor(() => expect(tree.toJSON()).toMatchSnapshot());
  });

  it('should list changed when navigation clicked', async () => {
    const mockNavigation = useNavigation();
    const { result: resultState } = renderHook(() => useState(0));
    const { result: resultForm } = renderHook(() => FormState.useState((s) => s.form));
    const { result: resultValues } = renderHook(() => FormState.useState((s) => s.currentValues));
    const [currentPage, setCurrentPage] = resultState.current;
    const { json: formJSON } = resultForm.current;
    const values = resultValues.current;

    const { getByTestId, rerender, debug } = render(
      <FormDataDetails
        navigation={mockNavigation}
        formJSON={formJSON}
        values={values}
        currentPage={currentPage}
        setCurrentPage={setCurrentPage}
      />,
    );

    const buttonNext = getByTestId('button-next');
    expect(buttonNext).toBeDefined();
    fireEvent.press(buttonNext);

    act(() => {
      setCurrentPage(1);
    });

    await waitFor(() => {
      const questionText = getByTestId('text-question-0');
      expect(questionText).toBeDefined();
      const answerText = getByTestId('text-answer-0');
      expect(answerText).toBeDefined();
      const paginationText = getByTestId('text-pagination');
      expect(paginationText).toBeDefined();
      expect(paginationText.props.children).toEqual([2, '/', 2]);
    });
  });

  it('should clear currentValues when back button clicked', async () => {
    const mockNavigation = useNavigation();
    const { result: resultState } = renderHook(() => useState(0));
    const { result: resultForm } = renderHook(() => FormState.useState((s) => s.form));
    const { result: resultValues } = renderHook(() => FormState.useState((s) => s.currentValues));
    const [currentPage, setCurrentPage] = resultState.current;
    const { json: formJSON } = resultForm.current;
    const values = resultValues.current;

    render(
      <FormDataDetails
        navigation={mockNavigation}
        formJSON={formJSON}
        values={values}
        currentPage={currentPage}
        setCurrentPage={setCurrentPage}
      />,
    );

    act(() => {
      triggerBeforeRemoveEvent({ data: { action: { type: 'FormData' } } });
      FormState.update((s) => {
        s.currentValues = {};
      });
    });

    await waitFor(() => {
      expect(resultValues.current).toEqual({});
    });
  });
});
