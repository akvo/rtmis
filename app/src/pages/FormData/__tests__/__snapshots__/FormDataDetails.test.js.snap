// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormDataDetails should match with snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#f9fafb",
      "flex": 1,
      "flexDirection": "column",
      "flexWrap": "wrap",
    }
  }
  testID="stack-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#f3f4f6",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "minheight": 78,
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="base-layout-page-title"
  >
    <RNCSafeAreaView
      edges={
        [
          "left",
          "top",
          "right",
        ]
      }
      style={
        {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "flex-start",
            "flex": 1,
          }
        }
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#242424",
            }
          }
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 3,
          }
        }
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#242424",
              "fontSize": 18,
              "paddingTop": 4,
              "textAlign": "center",
            }
          }
          testID="page-title"
        >
          Datapoint name
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "flex-end",
            "flex": 1,
          }
        }
      />
    </RNCSafeAreaView>
  </View>
  <RCTScrollView
    style={
      {
        "width": "100%",
      }
    }
  >
    <View>
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomWidth": 0.5,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
          testID="RNE__LISTITEM__padView"
        >
          <View
            style={
              [
                {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
            theme={
              {
                "colors": {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": {
                    "android": {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
          >
            <Text
              accessibilityRole="text"
              style={
                {
                  "backgroundColor": "transparent",
                  "color": "#242424",
                  "fontSize": 14,
                  "fontWeight": 700,
                  "marginBottom": 4,
                }
              }
              testID="text-question-0"
            />
            <Text
              accessibilityRole="text"
              style={
                {
                  "backgroundColor": "transparent",
                  "color": "#242424",
                  "fontSize": 15,
                }
              }
              testID="listItemTitle"
            >
              <Text
                testID="text-answer-0"
              >
                -
              </Text>
            </Text>
          </View>
        </View>
      </View>
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomWidth": 0.5,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
          testID="RNE__LISTITEM__padView"
        >
          <View
            style={
              [
                {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
            theme={
              {
                "colors": {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": {
                    "android": {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
          >
            <Text
              accessibilityRole="text"
              style={
                {
                  "backgroundColor": "transparent",
                  "color": "#242424",
                  "fontSize": 14,
                  "fontWeight": 700,
                  "marginBottom": 4,
                }
              }
              testID="text-question-1"
            />
            <Text
              accessibilityRole="text"
              style={
                {
                  "backgroundColor": "transparent",
                  "color": "#242424",
                  "fontSize": 15,
                }
              }
              testID="listItemTitle"
            >
              <Text
                testID="text-answer-1"
              >
                Test Village
              </Text>
            </Text>
          </View>
        </View>
      </View>
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomWidth": 0.5,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
          testID="RNE__LISTITEM__padView"
        >
          <View
            style={
              [
                {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
            theme={
              {
                "colors": {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": {
                    "android": {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
          >
            <Text
              accessibilityRole="text"
              style={
                {
                  "backgroundColor": "transparent",
                  "color": "#242424",
                  "fontSize": 14,
                  "fontWeight": 700,
                  "marginBottom": 4,
                }
              }
              testID="text-question-2"
            />
            <Text
              accessibilityRole="text"
              style={
                {
                  "backgroundColor": "transparent",
                  "color": "#242424",
                  "fontSize": 15,
                }
              }
              testID="listItemTitle"
            >
              <View
                testID="text-type-geo-2"
              >
                <Text>
                  Latitude
                  : 
                  -7.391187
                </Text>
                <Text>
                  Longitude
                  : 
                  109.4651083
                </Text>
              </View>
            </Text>
          </View>
        </View>
      </View>
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomWidth": 0.5,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
          testID="RNE__LISTITEM__padView"
        >
          <View
            style={
              [
                {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
            theme={
              {
                "colors": {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": {
                    "android": {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
          >
            <Text
              accessibilityRole="text"
              style={
                {
                  "backgroundColor": "transparent",
                  "color": "#242424",
                  "fontSize": 14,
                  "fontWeight": 700,
                  "marginBottom": 4,
                }
              }
              testID="text-question-3"
            />
            <Text
              accessibilityRole="text"
              style={
                {
                  "backgroundColor": "transparent",
                  "color": "#242424",
                  "fontSize": 15,
                }
              }
              testID="listItemTitle"
            >
              <Text
                testID="text-answer-3"
              >
                2021-12-22
              </Text>
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "white",
            "borderBottomColor": "silver",
            "borderLeftColor": "transparent",
            "borderRightColor": "transparent",
            "borderTopColor": "transparent",
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "column",
            "gap": 8,
            "padding": 16,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": 700,
              "marginBottom": 4,
            }
          }
          testID="text-question-4"
        />
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            onLoad={[Function]}
            source={
              {
                "uri": "data:image/jpeg;base64,dummyR0ndomb5e64",
              }
            }
            style={
              {
                "aspectRatio": 1,
                "bottom": 0,
                "height": 200,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": "100%",
              }
            }
            testID="image-answer-4"
            theme={
              {
                "colors": {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": {
                    "android": {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
            transitionDuration={360}
          />
          <View
            accessibilityElementsHidden={true}
            collapsable={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "aspectRatio": 1,
                  "backgroundColor": "#bdbdbd",
                  "height": 200,
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
              testID="RNE__Image__placeholder"
            />
          </View>
          <View
            style={
              {
                "aspectRatio": 1,
                "height": 200,
                "width": "100%",
              }
            }
            testID="RNE__Image__children__container"
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomWidth": 0,
        "borderTopColor": "grey",
        "borderTopWidth": 0.5,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 2,
          },
          undefined,
          false,
        ]
      }
      testID="RNE_BUTTON_WRAPPER"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": false,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="button-back"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 2,
              "borderWidth": 0,
              "flexDirection": "row",
              "fontSize": 14,
              "justifyContent": "center",
              "padding": 8,
              "paddingHorizontal": 8,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 12,
                },
                {
                  "color": "#9ca3af",
                  "fontSize": 18,
                  "paddingHorizontal": 4,
                },
                {
                  "fontFamily": "ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <Text
      style={
        {
          "color": "#1f2937",
          "fontSize": 14,
          "textAlign": "center",
          "width": "60%",
        }
      }
      testID="text-pagination"
    >
      1
      /
      2
    </Text>
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 2,
          },
          undefined,
          false,
        ]
      }
      testID="RNE_BUTTON_WRAPPER"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": false,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="button-next"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 2,
              "borderWidth": 0,
              "flexDirection": "row-reverse",
              "fontSize": 14,
              "justifyContent": "center",
              "padding": 8,
              "paddingHorizontal": 8,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 12,
                },
                {
                  "color": "#1f2937",
                  "fontSize": 18,
                  "paddingHorizontal": 4,
                },
                {
                  "fontFamily": "ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
